src/orca/services/nextflowtower/client.py

```diff
@@ -0,0 +1,128 @@
+            by the ``api_endpoints`` value.
+    """
+
+    auth_token: str
+    api_endpoint: str
+
+    def update_kwarg(
+        self, kwargs: dict[str, Any], key1: str, key2: str, default: Any
+    ) -> None:
+        """Ensure a default value for a nested key in kwargs.
+
+        Args:
+            kwargs: Keyword arguments
+            key1: Top-level key.
+            key2: Nested key.
+            default: Default value.
+        """
+        kwargs.setdefault(key1, dict())
+        if not isinstance(kwargs[key1], dict):
+            message = f"The '{key1}' keyword argument must be a dictionary."
+            raise ValueError(message)
+        kwargs[key1].setdefault(key2, default)
+        if not isinstance(kwargs[key1][key2], type(default)):
+            message = (
+                f"The value for kwargs['{key1}']['{key2}'] ({kwargs[key1][key2]}) "
+                f"is not the expected type ({type(default)})."
+            )
+            raise ValueError(message)
+
+    def request(self, method: str, path: str, **kwargs) -> requests.Response:
+        """Make an authenticated HTTP request.
+
+        Args:
+            method: An HTTP method (GET, PUT, POST, or DELETE).
+            path: The API path with the parameters filled in.
+            **kwargs: Additional named arguments passed through to
+                requests.request().
+
+        Returns:
+            The raw Response object to allow for special handling
+        """
+        valid_methods = {"GET", "PUT", "POST", "DELETE"}
+        if method not in valid_methods:
+            message = f"Method ({method}) not among valid options ({valid_methods})."
+            raise ValueError(message)
+
+        url = f"{self.api_endpoint}/{path}"
+
+        auth_header = f"Bearer {self.auth_token}"
+        self.update_kwarg(kwargs, "headers", "Authorization", auth_header)
+
+        return requests.request(method, url, **kwargs)
+
+    def request_json(self, method: str, path: str, **kwargs) -> dict[str, Any]:
+        """Make an authenticated HTTP request and parse the JSON response.
+
+        See ``TowerClient.request`` for argument definitions.
+
+        Raises:
+            HTTPError: If something went wrong with the request.
+
+        Returns:
+            A dictionary from deserializing the JSON response.
+        """
+        response = self.request(method, path, **kwargs)
+        response.raise_for_status()
+        return response.json()
+
+    def request_paged(self, method: str, path: str, **kwargs) -> list[dict[str, Any]]:
+        """Iterate through pages of results for a given request.
+
+        See ``TowerClient.request`` for argument definitions.
+
+        Raises:
+            ValueError: If the 'params' keyword argument
+                isn't a dictionary
+
+        Returns:
+            The cumulative list of items from all pages.
+        """
+        self.update_kwarg(kwargs, "params", "max", 50)
```

Comment:
I understand the reason for these two parameters and actually like the way you've implemented it, but what is your reason for not including `max` and `offset` as an argument (`def request_paged(...., max=50...)`) and as a kwarg?

Reply:
For consistency with the other `request` methods and for simplicity. For example, if I add a `max` parameter, I now need to handle (and document) the case where a user specifies `max` in the parameter and in the `params` dictionary. The user already has a way to provide this value, so I don't see the need to add another way, especially since I don't expect users to edit this value very often.

Reply:
+1